Add security v2 infrastructure and commands to libhoth

ashncook opened the pull request:
Add ability to get security information using security v2 commands from a given device using htool.

nnooney commented on examples/host_commands.h:
```diff
@@ -48,6 +49,68 @@ struct hoth_response_flash_spi_info {
+#define EC_CMD_BOARD_SPECIFIC_LAST 0x3FFF
+
+// NOTE: All further commands in this file are offset by
+// EC_CMD_BOARD_SPECIFIC_BASE.
```
I don't think these definitions are used by the security version command, let's remove this

nnooney commented on examples/host_commands.h:
```diff
@@ -48,6 +49,68 @@ struct hoth_response_flash_spi_info {
+ * identified by the "major_command" and "minor_command" fields of a command's
+ * request header.
+ */
+#define EC_PRV_CMD_HOTH_SECURITY_V2 0x0003
```
nit: HOTH_PRV_CMD_HOTH rather than EC_PRV_CMD_HOTH, to match other constants throughout this file

nnooney commented on examples/htool_security_version.h:
```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Google LLC
```
2025

cjevans-google commented:
General concerns:
* I overlooked this PR for several weeks; apologies.
* This is mixing some generic refactoring with the htool_get_security_version changes in the same PR, which is a tad odd to me.
* The github automatic rebase I'm not confident is correct; there's several removals in the headers that don't look like I expect.

ashncook commented:
> General concerns:
> 
> * I overlooked this PR for several weeks; apologies.
> * This is mixing some generic refactoring with the htool_get_security_version changes in the same PR, which is a tad odd to me.
> * The github automatic rebase I'm not confident is correct; there's several removals in the headers that don't look like I expect.

I have addressed your concerns.

This PR is ready for another round of reviews, please take another look at your convenience.
